crates/wasmtime/src/runtime/debug.rs

```diff
@@ -516,3 +550,196 @@ pub trait DebugHandler: Clone + Send + Sync + 'static {
         event: DebugEvent<'_>,
     ) -> impl Future<Output = ()> + Send;
 }
+
+/// Breakpoint state for modules within a store.
+#[derive(Default)]
+pub(crate) struct BreakpointState {
+    /// Single-step mode.
+    single_step: bool,
+    /// Breakpoints added individually.
+    breakpoints: BTreeSet<BreakpointKey>,
+}
+
+/// A breakpoint.
+pub struct Breakpoint {
+    /// Reference to the module in which we are setting the breakpoint.
+    pub module: Module,
+    /// Wasm PC offset within the module.
+    pub pc: u32,
```

Comment:
Not 100% necessary in this PR but: we might want to newtype this, just because we are often also talking about native PCs and text offsets in various places, and we don't want to mix them all up. (We should probably newtype those things too, IMO).